src/node_dependencies/resolve.js

```diff
@@ -42,20 +43,44 @@ const REQUEST_RESOLVE_MIN_VERSION = '8.9.0'
 // We need to use `new Promise()` due to a bug with `utils.promisify()` on
 // `resolve`:
 //   https://github.com/browserify/resolve/issues/151#issuecomment-368210310
-const resolvePathPreserveSymlinks = function (path, basedir) {
-  return new Promise((resolve, reject) => {
+const resolvePathPreserveSymlinksForDir = function (path, basedir) {
+  return new Promise((resolve) => {
     resolveLib(path, { basedir, preserveSymlinks: true }, (error, resolvedLocation) => {
       if (error) {
-        return reject(error)
+        resolve({ error })
       }
 
-      resolve(resolvedLocation)
+      resolve({ resolvedLocation })
     })
   })
 }
 
-const resolvePathFollowSymlinks = function (path, basedir) {
-  return require.resolve(path, { paths: [basedir] })
+// the resolve library has a `paths` option but it's not the same as multiple basedirs
+// see https://github.com/browserify/resolve/issues/188#issuecomment-679010477
+// we return the first resolved location or the first error if all failed
+const resolvePathPreserveSymlinks = async function (path, baseDirs) {
+  const tasks = baseDirs.map((basedir) => async (previousValue) => {
+    if (previousValue.resolvedLocation !== undefined) {
+      return previousValue
+    }
+    const { resolvedLocation, error } = await resolvePathPreserveSymlinksForDir(path, basedir)
+    if (resolvedLocation !== undefined) {
+      return { resolvedLocation }
+    }
+    // report the first error
+    return { error: previousValue.error || error }
+  })
+
+  const { resolvedLocation, error } = await pWaterfall(tasks, {})
+  if (resolvedLocation !== undefined) {
+    return resolvedLocation
+  }
+
+  throw error
```

Comment:
[sand] Would the following work?

Keeping the original `resolve` and `reject` from `resolvePathPreserveSymlinksForDir()` (so it throws on error) then:

```js
const resolvePathPreserveSymlinks = async function (path, baseDirs) {
  for (const baseDir of baseDirs) {
    const resolvedLocation = await resolvePathPreserveSymlinksForDir(path, basedir)
    if (resolvedLocation !== undefined) { 
      return resolvedLocation
    }
  }
}
```

Reply:
This should work:
```js
// eslint-disable-next-line fp/no-let
let firstError
// eslint-disable-next-line fp/no-loops
for (const basedir of baseDirs) {
  try {
    // eslint-disable-next-line no-await-in-loop
    return await resolvePathPreserveSymlinksForDir(path, basedir)
  } catch (error) {
    // eslint-disable-next-line fp/no-mutation
    firstError = firstError || error
  }
}

throw firstError
```

I took the first approach to follow our linting rules, but I think the one you suggested is easier to read. WDYT?

Reply:
Oh right, I forgot the first error is better reported than the later ones. Sounds good!

Reply:
Done in https://github.com/netlify/zip-it-and-ship-it/pull/302/commits/585c3e262aca5b856c107bc376550476e358d464